pkg/bridge/target/peer.go

```diff
@@ -0,0 +1,98 @@
+package target
+
+import (
+	"context"
+	"errors"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/ethpandaops/ethcore/pkg/execution/mimicry"
+)
+
+type Peer struct {
+	log logrus.FieldLogger
+
+	nodeRecord string
+
+	client *mimicry.Client
+
+	ready bool
+}
+
+func NewPeer(ctx context.Context, log logrus.FieldLogger, nodeRecord string) (*Peer, error) {
+	client, err := mimicry.New(ctx, log, nodeRecord, "mempool-bridge")
+	if err != nil {
+		return nil, err
+	}
+
+	p := Peer{
+		log:        log.WithField("node_record", nodeRecord),
+		nodeRecord: nodeRecord,
+		client:     client,
+		ready:      false,
+	}
+
+	return &p, nil
+}
+
+func (p *Peer) Start(ctx context.Context) (<-chan error, error) {
+	response := make(chan error)
+
+	p.client.OnStatus(ctx, func(ctx context.Context, status *mimicry.Status) error {
+		p.ready = true
+		return nil
+	})
+
+	p.client.OnDisconnect(ctx, func(ctx context.Context, reason *mimicry.Disconnect) error {
+		p.ready = false
+		str := "unknown"
+		if reason != nil {
+			str = reason.Reason.String()
+		}
+
+		p.log.WithFields(logrus.Fields{
+			"reason": str,
+		}).Debug("disconnected from client")
+
+		response <- errors.New("disconnected from peer (reason " + str + ")")
+
+		return nil
+	})
+
+	p.log.Debug("attempting to connect to client")
+
+	err := p.client.Start(ctx)
+	if err != nil {
+		p.log.WithError(err).Debug("failed to dial client")
+		return nil, err
+	}
+
+	return response, nil
+}
+
+func (p *Peer) Stop(ctx context.Context) error {
+	if p.client != nil {
+		if err := p.client.Stop(ctx); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *Peer) BridgeTransactions(ctx context.Context, transactions *mimicry.Transactions) error {
```

Comment:
Nitpicky but i suppose this could be closer to like `SendTransactions`? This peer struct doesn't really know about a bridge. The package above this might be `SendTransactionsToPeers` and the `bridge` package the layer above that could be `Broadcast(ctx, transactions` or something along those lines? Not a huge deal

Reply:
:woman_artist: